### PR TITLE
Use hostname check from verify.go to handle patterns in TLS certs

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

Previously, DNSNames in x509 certs with wildcards were not accepted. The function from Golang's `verify.go` is taken, so the check remains the same between Golang versions.

## Why is it important?

Wildcards in DNSNames were not accepted. Now the are.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

**Comment by michalpristas** [comment]
_testing_

File: libbeat/common/transport/tlscommon/validhostname.go [hunk 1]
```diff
@@ -47,7 +47,69 @@
...
+		}
+	}
+
+	return true
+}
+
+// toLowerCaseASCII returns a lower-case version of in. See RFC 6125 6.4.1. We use
+// an explicitly ASCII function to avoid any sharp corners resulting from
+// performing Unicode operations on DNS labels.
+func toLowerCaseASCII(in string) string {
```
Comment (mdelapenya) [inline libbeat/common/transport/tlscommon/validhostname.go:85]: It seems fairly easy to add unit tests for for these two methods. Could we add them too?

Comment (kvch) [reply libbeat/common/transport/tlscommon/validhostname.go:85]: I am adding tests ATM, but not for these specific function, but for the whole VerifyConnection callback.

**Review by michalpristas — APPROVED** [review]
worked for agent enroll

File: libbeat/common/transport/tlscommon/validhostname.go (hunk 1 above)
Comment (mdelapenya) [reply libbeat/common/transport/tlscommon/validhostname.go:85]: Great, thanks! If possible, I'd add unit tests for the calculation and verification of the lowerCaseASCII hostname, just in case we missed any edge case for the conditional branches inside both methods.

**Comment by kvch** [comment]
jenkins run tests

**Comment by kvch** [comment]
Failing test are unrelated.
